### PR TITLE
Add fill resize manipulator

### PR DESCRIPTION
**Pull request by jcarouth**
First stab at customizing the resize manipulator in a semi sane way.
I'm not too excited about having to manually construct the Server
instead of using the ServerFactory, but at a quick glance it didn't
look like it was too easy to override the default too easily.

### CODE REVIEW

File: src/liftopia/Manipulator/FillCanvasSize.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        if (round($width) !== round($image->width()) or
+            round($height) !== round($image->height())) {
+            $image = $this->runResize($image, $fit, round($width), round($height), $crop, $bgColor);
+        }
+
+        return $image;
+    }
+
+    public function getBgColor($bgColor)
```
Comment (jakefolio) [inline src/liftopia/Manipulator/FillCanvasSize.php:43]: Missing docblocks

File: src/liftopia/Manipulator/FillCanvasSize.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        $hexColorValidationRegex = '/#([a-f]|[A-F]|[0-9]){3}(([a-f]|[A-F]|[0-9]){3})?\b/';
+
+        if (!preg_match($hexColorValidationRegex, $bgColor)) {
+            return null;
+        }
+
+        return $bgColor;
+    }
+
+    public function getFit($fit)
```
Comment (jakefolio) [inline src/liftopia/Manipulator/FillCanvasSize.php:65]: Missing docblocks

File: src/liftopia/Manipulator/FillCanvasSize.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        if ($fit === 'fill') {
+            return 'fill';
+        }
+
+        return parent::getFit($fit);
+    }
+
+    public function runResize(Image $image, $fit, $width, $height, $crop = null, $bgColor = '#ffffff')
```
Comment (jakefolio) [inline src/liftopia/Manipulator/FillCanvasSize.php:85]: Missing docblocks

File: src/liftopia/Manipulator/FillCanvasSize.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        if ($fit === 'fill') {
+            return $this->runFillResize($image, $width, $height, $bgColor);
+        }
+
+        return parent::runResize($image, $fit, $width, $height, $crop);
+    }
+
+    public function runFillResize(Image $image, $width, $height, $bgColor)
```
Comment (jakefolio) [inline src/liftopia/Manipulator/FillCanvasSize.php:103]: Missing docblocks

File: src/liftopia/Manipulator/FillCanvasSize.php [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        $width = $this->getWidth($request->get('w'));
+        $height = $this->getHeight($request->get('h'));
+        $fit = $this->getFit($request->get('fit'));
+        $crop = $this->getCrop($request->get('crop'));
+        $bgColor = $this->getBgColor($request->get('bg'));
+
+        list($width, $height) = $this->resolveMissingDimensions($image, $width, $height);
+        list($width, $height) = $this->limitImageSize($width, $height);
+
+        if (round($width) !== round($image->width()) or
```
Comment (jakefolio) [inline src/liftopia/Manipulator/FillCanvasSize.php]: Are we using "or" or || in conditionals?

Comment (jcarouth) [inline src/liftopia/Manipulator/FillCanvasSize.php]: We use `||`. But I copypastad this from the Glide Size manipulator so I could hack in the $bgColor thing.

**Comment by jakefolio** [comment]
This PR is super ceded by PR #2.

**Pull request closed without merging**